### PR TITLE
feat: add support for labels when deploying a pipeline

### PR DESCRIPTION
**Pull request by jLopezbarb**
# Proposed changes

Fixes #6996

Adds CLI support for labels flag under the pipeline command. If the okteto cluster that the user is not connected doesn't support the labels it returns a user friendly error as you can see below:
Captura de pantalla 2023-09-05 a las 13 53 12

## How to validate

1. Use a cluster context that supports pipeline labels 
1. Run `okteto pipeline deploy --branch cli-e2e --repository https://github.com/okteto/movies --namespace your-ns --wait --label=key`
1. Check that the configmap generated has the label `key`

## CLI Quality Reminders 🔧

For both authors and reviewers:

- Scrutinize for potential regressions
- Ensure key automated tests are in place
- Build the CLI and test using the validation steps
- Assess Developer Experience impact (log messages, performances, etc)
- If too broad, consider breaking into smaller PRs
- Adhere to our [code style](https://github.com/okteto/okteto/blob/master/docs/code-style.md) and [code review](https://github.com/okteto/okteto/blob/master/docs/code-review.md) guidelines

### CODE REVIEW

File: cmd/pipeline/deploy.go [hunk 1]
```diff
@@ -124,6 +126,7 @@ func deploy(ctx context.Context) *cobra.Command {
 	if err := cmd.Flags().MarkHidden("filename"); err != nil {
 		oktetoLog.Infof("failed to mark 'filename' flag as hidden: %s", err)
 	}
+	cmd.Flags().StringArrayVarP(&flags.labels, "label", "", []string{}, "set a preview environment label (can be set more than once)")
```
Comment (teresaromero) [inline cmd/pipeline/deploy.go:129]:
```suggestion
	cmd.Flags().StringArrayVarP(&flags.labels, "label", "", []string{}, "set an environment label (can be set more than once)")
```

File: pkg/okteto/pipeline.go [hunk 2]
```diff
@@ -26,6 +27,14 @@ import (
 	"github.com/shurcooL/graphql"
 )
 
+var (
+	ErrDeployPipelineLabelsFeatureNotSupported = errors.New(`deploying a preview environments with labels requires a more recent version of Okteto.
```
Comment (teresaromero) [inline pkg/okteto/pipeline.go:31]:
```suggestion
	ErrDeployPipelineLabelsFeatureNotSupported = errors.New(`deploying an environment with labels requires a more recent version of Okteto.
```

File: pkg/okteto/pipeline.go [hunk 3]
```diff
@@ -104,91 +117,83 @@ type deprecatedDestroyPipelineResponse struct {
...
-			ID:     string(mutation.Response.Action.Id),
-			Name:   string(mutation.Response.Action.Name),
-			Status: string(mutation.Response.Action.Status),
-		}
-		gitDeployResponse.GitDeploy = &types.GitDeploy{
-			ID:         string(mutation.Response.GitDeploy.Id),
-			Name:       string(mutation.Response.GitDeploy.Name),
-			Repository: string(mutation.Response.GitDeploy.Repository),
-			Status:     string(mutation.Response.GitDeploy.Status),
+	gitDeployResponse := &types.GitDeployResponse{
```
Comment (teresaromero) [inline pkg/okteto/pipeline.go:143]: aren't we returning the labels as field?

**Review by teresaromero — APPROVED** [review]
Tested with supporting version and not supported version. 👍🏻 

Minor comments about typos, approving when fixed

File: pkg/okteto/pipeline.go (hunk 3 above)
Comment (jLopezbarb) [reply pkg/okteto/pipeline.go:143]: no, We just return the action ID and git deploy which are the necessary fields in order to show logs and know if it's been deployed.

**Review by andreafalzetti — APPROVED** [review]
Tested and works as expected ✅

File: pkg/okteto/pipeline.go [hunk 4]
```diff
@@ -104,91 +117,83 @@ type deprecatedDestroyPipelineResponse struct {
...
-			"space":      graphql.String(opts.Namespace),
-			"branch":     graphql.String(opts.Branch),
-			"variables":  variablesArg,
-			"filename":   graphql.String(opts.Filename),
-		}
 
-		err := mutate(ctx, &mutation, queryVariables, c.client)
+	mutationVariables := c.getDeployVariables(opts)
+	var response deployPipelineResponse
+	if len(opts.Labels) == 0 {
```
Comment (AdrianPedriza) [inline pkg/okteto/pipeline.go:123]: cannot we use just one mutation struct but if no labels send labels empty?

File: pkg/okteto/pipeline.go [hunk 5]
```diff
@@ -26,6 +27,14 @@ import (
 	"github.com/shurcooL/graphql"
 )
 
+var (
+	ErrDeployPipelineLabelsFeatureNotSupported = errors.New(`deploying a preview environments with labels requires a more recent version of Okteto.
+
+    Consider removing the "--label" flag, or please upgrade to the latest version.
+
+    For more information and upgrade instructions, please visit our docs at https://www.okteto.com/docs or contact your system administrator.`)
```
Comment (AdrianPedriza) [inline pkg/okteto/pipeline.go:35]: I think this is mora a hint than an error. Should we move as hint?

File: cmd/pipeline/deploy.go [hunk 6]
```diff
@@ -124,6 +126,7 @@ func deploy(ctx context.Context) *cobra.Command {
 	if err := cmd.Flags().MarkHidden("filename"); err != nil {
 		oktetoLog.Infof("failed to mark 'filename' flag as hidden: %s", err)
 	}
+	cmd.Flags().StringArrayVarP(&flags.labels, "label", "", []string{}, "set an environment label (can be set more than once)")
```
Comment (andreafalzetti) [inline cmd/pipeline/deploy.go:129]: In the `pipeline list` I am using this description:

```
"tag and organize dev environments using labels (multiple --label flags accepted)"
```

Which I suggested at the time for `preview list`. I think it helps from a product stand point to understand what `--label` is meant for.

**Pull request merged**